firefox-ios/README.md

```diff
@@ -80,4 +80,82 @@ All developers must follow the following Swift style to keep the code orginized:
+general detailed code structure can be found at (note: this code structure are for windows, but its the same for all platforms in general):
+* https://developer.mozilla.org/en-US/docs/Mozilla/Developer_guide/Source_Code/Directory_structure
+* http://codefirefox.com/video/source-structure
+
+![](code structor.png)
+
+[Reference](https://github.com/mozilla/firefox-ios)
+
+
+## Stakeholder view
+* **Communicators**:
+
+ are the people that provide documentation for the users and administrators of the system.
+ Furthermore they may provide training for staff or the development team.
+ Often communicators promote the product by communicating the product's key features and benefits to other stakeholders.
+
+ Firefox is a global webbrowser, They have a "Mobilizers" community. Mobilizers are tech/geek enthusiasts that are committed to spreading Firefox OS, Mozilla's new mobile product. Mobilizers receive periodic missions asking them to do a simple action to teach others about the new Firefox OS. Mobilizers will also attend, assist or host events in their market about once a month [Reference](https://wiki.mozilla.org/FirefoxOS/Community/Mobilizers).
+
+* **Developers**:
+
+ are responsible for the implementation and deployment from the systems' specification.
+ That is: design, code, test and accept.
+ Their concerns lie with the platform used, programming languages for writing code, tools used, dependencies needed, as well as the maintainability and flexibility of the system.
+
+ The most active and notable developers at this moment are: Stefan Arentz, Brian Nicholson, Stephan Leroux and Richard Newman. These developers create pull requests with new features or fixes (Stefan Arentz does only fixes for bugs). It is important to note that these are not the only developers working on the firefox project. The most core team can be found at [Firefox-team](https://wiki.mozilla.org/Firefox/Team/whois).
+
+
+* **Testers**:
+
+ test the system to ensure that it works exactly the way it was intended to work.
+ 
+ [Stephan Leroux](https://github.com/sleroux) and [Brian Nicholson](https://github.com/thebnich), contribute the most to the tests, code tests, UI tests, etc... 
+
+* **Funders**:
+
+ Funders give money to the project to keep it alive.
+ 
+ One of the best money incoming for firefox are the search engines such as Google, who pays firefox to make Google search engine as the default search engine for firefox.
+ 
+* **General stakeholders**:
+ * Product Manager: Jen Bertsch
+ * Project Manager: Mike Alexis
+ * Content owner: Greg Jost
+ * Copy: Natalie Linden
+ * UX and Design: ZURB
+ * Creative lead: Tim Murray
+ * Foundation stakeholder: Andrea Wood
+ * Advocacy stakeholder: Jochai Ben-Avie
+ * Legal: Mika Devi
+ * Web dev: Craig Cook
+ * Analytics: Gareth C
+ * L10N: Flod, Pascal Chevrel
+ 
+ [Reference](https://wiki.mozilla.org/Websites/Mozilla.org/Smart_On)
+ 
+ ## UML - Sequence diagram
```

Comment:
Beneath the image there is the source for the diagram, and inside the image there is a url to the tool that was used.